tasks/init.rb

```diff
@@ -0,0 +1,62 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+require 'facter'
+require 'json'
+
+params   = JSON.parse(STDIN.read)
+timeout  = params['timeout'].to_i
+message = params['messsage'] || nil
+
+case Facter.value(:kernel)
+when 'windows'
+  def shutdown_command
+    if File.exists?("#{ENV['SYSTEMROOT']}\\sysnative\\shutdown.exe")
+      "#{ENV['SYSTEMROOT']}\\sysnative\\shutdown.exe"
+    elsif File.exists?("#{ENV['SYSTEMROOT']}\\system32\\shutdown.exe")
+      "#{ENV['SYSTEMROOT']}\\system32\\shutdown.exe"
+    else
+      'shutdown.exe'
+    end
+  end
+
+  def async_shutdown(shutdown_cmd)
```

Comment:
Not sure how I feel about dynamically creating functions in a case.  This may make testing very difficult.